### PR TITLE
Drop node.js v6 support

### PR DESCRIPTION
**Pull request by mojosoeun**
### Description of the Change

According to this issue(#3884), Node.js v6 will be the end of life on April 30th. 
So I removed v6 and changed md file.

### Alternate Designs

N/A

### Why should this be in core?

It's minimum Node.js version.

### Benefits

Mocha doesn't care issues with Node.js v6 anymore.

### Possible Drawbacks

N/A

### Applicable issues

close(#3884)

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -98,7 +98,7 @@ or as a development dependency for your project:
 $ npm install --save-dev mocha
 ```
 
-> As of v6.0.0, Mocha requires Node.js v6.0.0 or newer.
+> As of v6.0.0, Mocha requires Node.js v8.0.0 or newer.
```
Comment (Bamieh) [inline docs/index.md]: This is going to be a breaking change so `As of mocha v7.0.0`

Comment (outsideris) [reply docs/index.md]: Agreed.

Comment (mojosoeun) [reply docs/index.md]: Okay :) I fixed it!

File: docs/index.md [hunk 2]
```diff
@@ -98,7 +98,7 @@ or as a development dependency for your project:
 $ npm install --save-dev mocha
 ```
 
-> As of v6.0.0, Mocha requires Node.js v8.0.0 or newer.
+> As of v6.0.0, Mocha requires Node.js v7.0.0 or newer.
```
Comment (juergba) [inline docs/index.md]: Is this correct? Should be:
"As of v7.0.0, Mocha requires Node.js v8.0.0 or newer."
Next major Mocha release will be v7. And Node will be supported >=v8.

**Comment by boneskull** [comment]
This is a good opportunity to think more about our branching strategy.  If we merge this, then we cannot just cut a release from `master`.

I think we should discuss this in #3819.

**Comment by juergba** [comment]
@mojosoeun could you rebase, please. Probably there will also be conflicts to solve.
I would like to merge this PR, thanks!

**Comment by juergba** [comment]
merged, but `Smoke` still runs with node v6.

**Pull request merged**